org.eclipse.tm4e.ui/src/main/java/org/eclipse/tm4e/ui/internal/utils/PreferenceUtils.java

```diff
@@ -32,7 +36,14 @@ public final class PreferenceUtils {
 	public static boolean isDarkEclipseTheme() {
 		final IEclipsePreferences prefs = getE4PreferenceStore();
 		final String eclipseThemeId = prefs == null ? null : prefs.get(PreferenceConstants.E4_THEME_ID, null);
-		return eclipseThemeId != null && eclipseThemeId.toLowerCase().contains("dark");
+
+		if (eclipseThemeId != null) {
+			return eclipseThemeId.toLowerCase().contains("dark");
+		}
+
+		// If the default Eclipse theme is active, return the system's dark mode setting
+		return DEFAULT_THEME == PlatformUI.getWorkbench().getThemeManager().getCurrentTheme().getId() //
```

Comment:
Wouldn't it be more reliable to query something like `PlatformUI.getActiveWorkbenchWindow().getShell().getBackground()` here to establish whether the theme is dark or not?

Reply:
I think this is a good idea, we could use something like here to decide if it counts as a dark color:

https://github.com/eclipse/tm4e/blob/945f958d43469c1be511e90d63c91cf6787f13b3/org.eclipse.tm4e.ui/src/main/java/org/eclipse/tm4e/ui/internal/themes/AbstractThemeManager.java#L114-L118